pytest_plugins/auto_vault.py

```diff
@@ -0,0 +1,19 @@
+"""Plugin enables pytest to notify and update the requirements"""
+import os
+import re
+import subprocess
+from pathlib import Path
+
+
+def pytest_addoption(parser):
+    """Options to allow user to update the requirements"""
+    envpath = Path('.env')
```

Comment:
If the user runs the tests from other than the root directory, pytest fails with:
```$ pwd
/home/ogajduse/repos/robottelo/tests/foreman/cli

$ pytest --collect-only test_hammer.py 
Traceback (most recent call last):
  File "/home/ogajduse/.virtualenvs/robottelo/bin/pytest", line 8, in <module>
    sys.exit(console_main())
             ^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 189, in console_main
    code = main()
           ^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 147, in main
    config = _prepareconfig(args, plugins)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 328, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_hooks.py", line 265, in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_manager.py", line 80, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_callers.py", line 55, in _multicall
    gen.send(outcome)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/helpconfig.py", line 103, in pytest_cmdline_parse
    config: Config = outcome.get_result()
                     ^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_callers.py", line 39, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 1075, in pytest_cmdline_parse
    self.parse(args)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 1425, in parse
    self._preparse(args, addopts=addopts)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 1327, in _preparse
    self.hook.pytest_load_initial_conftests(
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_hooks.py", line 265, in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_manager.py", line 80, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_callers.py", line 60, in _multicall
    return outcome.get_result()
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_callers.py", line 39, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 1153, in pytest_load_initial_conftests
    self.pluginmanager._set_initial_conftests(
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 564, in _set_initial_conftests
    self._try_load_conftest(anchor, importmode, rootpath)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 581, in _try_load_conftest
    self._getconftestmodules(anchor, importmode, rootpath)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 610, in _getconftestmodules
    mod = self._importconftest(conftestpath, importmode, rootpath)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 658, in _importconftest
    self.consider_conftest(mod)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 740, in consider_conftest
    self.register(conftestmodule, name=conftestmodule.__file__)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 495, in register
    self.consider_module(plugin)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 748, in consider_module
    self._import_plugin_specs(getattr(mod, "pytest_plugins", []))
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 755, in _import_plugin_specs
    self.import_plugin(import_spec)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 792, in import_plugin
    self.register(mod, modname)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/_pytest/config/__init__.py", line 488, in register
    ret: Optional[str] = super().register(plugin, name)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in register
    hook._maybe_apply_history(hookimpl)
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_hooks.py", line 300, in _maybe_apply_history
    res = self._hookexec(self.name, [method], kwargs, False)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_manager.py", line 80, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_callers.py", line 60, in _multicall
    return outcome.get_result()
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/ogajduse/.virtualenvs/robottelo/lib/python3.11/site-packages/pluggy/_callers.py", line 39, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ogajduse/repos/robottelo/pytest_plugins/auto_vault.py", line 13, in pytest_addoption
    re.search(r'\s*#.*VAULT_SECRET_ID_FOR_DYNACONF', envpath.read_text())
                                                     ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/pathlib.py", line 1058, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/pathlib.py", line 1044, in open
    return io.open(self, mode, buffering, encoding, errors, newline)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '.env'
```

Please find a way to always point to the correct location.

Reply:
@ogajduse Yeah had that in mind but forgot to update the code. Thanks for catching !